cdap-watchdog/src/main/java/co/cask/cdap/logging/appender/remote/RemoteLogAppender.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.logging.appender.remote;
+
+
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.common.http.DefaultHttpRequestConfig;
+import co.cask.cdap.common.internal.remote.RemoteClient;
+import co.cask.cdap.common.service.RetryStrategies;
+import co.cask.cdap.logging.appender.AbstractLogPublisher;
+import co.cask.cdap.logging.appender.LogAppender;
+import co.cask.cdap.logging.appender.LogMessage;
+import co.cask.cdap.logging.appender.kafka.LogPartitionType;
+import co.cask.cdap.logging.serialize.LogSchema;
+import co.cask.cdap.logging.serialize.LoggingEventSerializer;
+import co.cask.common.http.HttpMethod;
+import co.cask.common.http.HttpRequest;
+import co.cask.common.http.HttpResponse;
+import com.google.common.hash.Hashing;
+import com.google.common.net.HttpHeaders;
+import com.google.inject.Inject;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericDatumWriter;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.DatumWriter;
+import org.apache.avro.io.Encoder;
+import org.apache.avro.io.EncoderFactory;
+import org.apache.twill.discovery.DiscoveryServiceClient;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.nio.ByteBuffer;
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Remote log appender to push logs to log saver.
+ */
+public class RemoteLogAppender extends LogAppender {
+  private static final String APPENDER_NAME = "RemoteLogAppender";
+  private final RemoteClient remoteClient;
+  private final RemoteLogPublisher publisher;
+
+  @Inject
+  public RemoteLogAppender(CConfiguration cConf, DiscoveryServiceClient discoveryServiceClient) {
+    setName(APPENDER_NAME);
+    int queueSize = cConf.getInt(Constants.Logging.APPENDER_QUEUE_SIZE, 512);
```

Comment:
since the cConf is already being passed to RemoteLogPublisher, this lookup can be done there along with everything else.